parse_cutadapt_logs.py

```diff
@@ -0,0 +1,174 @@
+
+    # Throw exception if no logfiles matched string.
+    if len(logfiles) == 0:
+        raise Exception("No logfiles found in folder " +
+                        args.input + " matching string " + args.match)
+
+    # Open output file for writing.
+    outfile = open(args.output, "w")
+
+    # Put samples in set to double-check they are all unique.
+    past_samples = set()
+
+    # Set flag for whether header printed or not.
+    header_printed = False
+
+    # Loop over all logfiles.
+    for logfile in logfiles:
+
+        sample = logfile.split(args.sample_delim)[0]
+
+        # Check whether sample already seen and if not add to set.
+        if sample in past_samples:
+            raise Exception("Parsed sample name " + sample +
+                            " is not unique!")
+
+        past_samples.add(sample)
+
+        log = args.input + "/" + logfile
+
+        # Initialize library object which will be either "single" or "paired"
+        library = None
+
+        # Initialize empty dict which will be actual counts parsed.
+        read_counts = {}
+
+        with open(log, "r") as infile:
+
+            for line in infile:
+
+                # Check if library defined. If not then check if library info
+                # is on line. If not then skip line. Set strings that should
+                # be matched as well depending on library type and what
+                # categories these strings correspond to.
+                if not library:
+                    if "in paired-end mode ..." in line:
+
+                        library = "paired"
+
+                        strings2match = ["Total read pairs processed:",
+                                         "Read 1 with adapter:",
+                                         "Read 2 with adapter:",
+                                         "Pairs written (passing filters):"]
+
+                        count_categories = ["pairs_in", "read1_match",
+                                            "read2_match", "pairs_passed"]
+
+                    elif "in single-end mode ..." in line:
+
+                        library = "single"
+
+                        strings2match = ["Total reads processed:",
+                                         "Reads with adapters:",
+                                         "Reads written (passing filters):"]
+
+                        count_categories = ["reads_in", "reads_match",
+                                            "reads_passed"]
+
+                    # Check if header printed yet and if not then do so.
+                    if library and not header_printed:
+                        header_out = ["sample"] + count_categories
+                        print("\t".join(header_out), file=outfile)
+                        header_printed = True
+
+                    # Skip to next line.
+                    continue
+
+                # If library defined then check for matching strings.
+                for i in range(len(strings2match)):
```

Comment:
To keep the `strings2match` and `count_categories` together without using a `range()`, you can use `zip()`:

```python
for string2match, count_category in zip(strings2match, count_categories):
    # ...
    pass
```